test/login.js

```diff
@@ -142,4 +158,59 @@ describe('Login', function() {
     });
   });
 
+  it('should log a whitelisted user', function(done) {
+    var user = apos.users.newInstance();
+
+    user.firstName = 'Admin';
+    user.lastName = 'Test';
+    user.title = 'Admin Test';
+    user.username = 'admin-test';
+    user.password = 'crookshanks';
+    user.email = 'admintest@aol.com';
+    user.lastLogin = new Date();
+    user.groupIds = [ apos.users.options.groups[1]._id ]; // admin group
+
+    apos.users.insert(apos.tasks.getReq(), user, function(err) {
+      assert(!err);
+      return request.post('http://localhost:7901/login', {
+        form: { username: 'admin-test', password: 'crookshanks' },
+        followAllRedirects: true,
+        jar: loginLogoutJar
+      }, function(err, response, body) {
+        assert(!err);
+        assert.equal(response.statusCode, 200);
+        assert(body.match(/logout/));
+        return done();
+      });
+    });
+  });
+
+  it('should disable an inactive user', function(done) {
+    var user = apos.users.newInstance();
+
+    user.firstName = 'Random';
+    user.lastName = 'Test';
+    user.title = 'Random Test';
+    user.username = 'random-test';
+    user.password = 'crookshanks';
+    user.email = 'randomtest@aol.com';
+    user.lastLogin = new Date();
+    user.groupIds = [ apos.users.options.groups[0]._id ]; // guest group
+
+    apos.users.insert(apos.tasks.getReq(), user, function(err) {
+      assert(!err);
+      return request.post('http://localhost:7901/login', {
+        form: { username: 'random-test', password: 'crookshanks' },
+        followAllRedirects: true,
+        jar: loginLogoutJar
+      }, function(err, response, body) {
+        assert(!err);
+        assert.equal(response.statusCode, 500);
```

Comment:
Don't return `500` for something that is not a failure of the system. Use an appropriate 400 series status code.

Reply:
If we want to inform the user its accound has been disabled, we need to override the traditional error message "Login or password incorrect". I did not succeed. I wanted to discuss that with you to check if you know the right move to make here. Meanwhile, I put this to have something working (I will update the test afterwards).

Reply:
Ok with custom callback from `passport`. I updated the test too